ash/src/device.rs

```diff
@@ -828,14 +832,15 @@ impl Device {
         &self,
         info: &vk::ImageSparseMemoryRequirementsInfo2,
     ) -> usize {
-        let mut count = 0;
+        // TODO: when pSparseMemoryRequirements is NULL, is it okay/safe for count to be uninit?
+        let mut count = mem::MaybeUninit::uninit();
```

Comment:
I think this is sound as rwitten. The spec seems to be missing the relevant text, but consider this analogous section:

> **If pPhysicalDevices is NULL**, then the number of physical devices available is returned in pPhysicalDeviceCount. Otherwise, pPhysicalDeviceCount must point to a variable set by the user to the number of elements in the pPhysicalDevices array [...]

Reply:
Oh, I thought I had deleted all TODOs from this PR again :sweat_smile:

Reply:
@Ralith I'm marking this as unresolved because, at least for `vkGetLatencyTimingsNV` (https://github.com/ash-rs/ash/issues/814, https://github.com/KhronosGroup/Vulkan-Docs/issues/2269#issuecomment-1813339455), `pTimingCount` will be set to the available number of timings _if it was `0`_, even if `pTimings` **was non-`NULL`**. Meaning that there's probably an `if (!pTimings || !*pTimingCount)` in the driver which could read uninitialized values.

@chansen-nv is adding that mention to the spec, but I don't think this behaviour is consistent/expected wrt other Vulkan functions?

---

However, https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkGetDeviceImageSparseMemoryRequirements.html only says:

> `pSparseMemoryRequirementCount` is a pointer to an integer related to the number of sparse memory requirements available or queried, as described below.

(There is no description below)

And the VUID doesn't rule out a similar implementation to `vkGetLatencyTimingsNV` either:

> If the value referenced by `pSparseMemoryRequirementCount` is not `0`, and `pSparseMemoryRequirements` is not `NULL`, `pSparseMemoryRequirements` must be a valid pointer to an array of `pSparseMemoryRequirementCount` [VkSparseImageMemoryRequirements2](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkSparseImageMemoryRequirements2.html) structures

Perhaps @oddhack can help us clarify whether or not the count pointer/value should be written when it's zero even if the array-pointer is `NULL` (in the generic case). If we cannot get a guarantee on this for some/most functions, we should back this change out and keep a `0` init (`let mut count = 0;`).